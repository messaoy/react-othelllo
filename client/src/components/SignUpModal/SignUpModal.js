import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import axios from 'axios';


export class SignUpModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      name: '',
      redirectToUser: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSignUp(user) {
    const { toggleModalSignUp } = this.props;
    axios.post('http://localhost:4001/api/users', user)
      .then(() => {
        this.setState({ redirectToUser: true });
      })
      .catch(() => {
        toggleModalSignUp(false);
      });
  }

  handleSubmit(e) {
    e.preventDefault();
    const { email, name } = this.state;
    const user = {
      email,
      name,
    };

    this.setState({ email: '', name: '' });
    return this.handleSignUp(user);
  }

  render() {
    const {
      email, name, redirectToUser,
    } = this.state;
    const { handleChange, handleSubmit } = this;
    const { toggleModalSignUp } = this.props;
    if (redirectToUser === true) {
      return <Redirect to="/" />;
    }
    return (
      <div id="modalSignUp">
        <div />
        <div>
          <header>
            <p>Inscription</p>
          </header>
          <div>
            <form>
              <div className="field">
                <p>
                  <input
                    id="emailInput"
                    required
                    className="input"
                    type="email"
                    name="email"
                    placeholder="Email"
                    onChange={handleChange}
                    value={email}
                  />
                  <span />
                </p>
              </div>
              <div className="field">
                <p>
                  <input
                    id="nameInput"
                    required
                    className="input"
                    type="text"
                    name="name"
                    placeholder="Name"
                    onChange={handleChange}
                    value={name}
                  />
                  <span />
                </p>
              </div>
              <footer>
                <button
                  id="submitButton"
                  type="submit"
                  onClick={handleSubmit}
                  className="button is-success"
                >
                  S&lsquo;inscrire
                </button>
                <button
                  id="cancelButton"
                  type="button"
                  onClick={() => toggleModalSignUp(false)}
                  className="button"
                >
                  Annuler
                </button>
              </footer>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default SignUpModal;
