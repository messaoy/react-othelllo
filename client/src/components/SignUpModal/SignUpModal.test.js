import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SignUpModal } from './SignUpModal';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  const toggleModalSignUp = jest.fn();
  const connect = jest.fn();
  wrapper = shallow(<SignUpModal toggleModalSignUp={toggleModalSignUp} connect={connect} />);
});

describe('Modal signUp tests', () => {
  it('Should exist', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper.exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have id #modalLogin', () => {
    const container = wrapper.find('#modalSignUp');
    expect(container).toHaveLength(1);
  });

  it('Should have a submit button', () => {
    const container = wrapper.find('#submitButton');
    expect(container).toHaveLength(1);
  });

  it('Should change email state with handle change on emailInput', () => {
    const input = wrapper.find('#emailInput');
    const mockEvent = {
      target: {
        name: 'email',
        value: 'test',
      },
    };
    input.simulate('change', mockEvent);
    expect(wrapper.state().email).toEqual('test');
  });

  it('Should change name state with handle change on nameInput', () => {
    const input = wrapper.find('#nameInput');
    const mockEvent = {
      target: {
        name: 'name',
        value: 'test',
      },
    };
    input.simulate('change', mockEvent);
    expect(wrapper.state().name).toEqual('test');
  });

  it('Should have cancel button', () => {
    const container = wrapper.find('#cancelButton');
    expect(container).toHaveLength(1);
  });

  it('Should call toggleModal() props on cancel button click', () => {
    const button = wrapper.find('#cancelButton');
    const toggle = wrapper.instance().props.toggleModalSignUp;
    button.simulate('click');
    expect(toggle).toHaveBeenCalled();
  });
});
