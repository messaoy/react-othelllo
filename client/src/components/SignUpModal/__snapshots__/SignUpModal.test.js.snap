// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal signUp tests should match the snapshot 1`] = `
<div
  id="modalSignUp"
>
  <div />
  <div>
    <header>
      <p>
        Inscription
      </p>
    </header>
    <div>
      <form>
        <div
          className="field"
        >
          <p>
            <input
              className="input"
              id="emailInput"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
            <span />
          </p>
        </div>
        <div
          className="field"
        >
          <p>
            <input
              className="input"
              id="nameInput"
              name="name"
              onChange={[Function]}
              placeholder="Name"
              required={true}
              type="text"
              value=""
            />
            <span />
          </p>
        </div>
        <footer>
          <button
            className="button is-success"
            id="submitButton"
            onClick={[Function]}
            type="submit"
          >
            Sâ€˜inscrire
          </button>
          <button
            className="button"
            id="cancelButton"
            onClick={[Function]}
            type="button"
          >
            Annuler
          </button>
        </footer>
      </form>
    </div>
  </div>
</div>
`;
